File: backend/src/main/java/com/bubbleflow/service/OrderService.java
package com.bubbleflow.service;

import com.bubbleflow.dto.order.request.OrderRequest;
import com.bubbleflow.dto.order.request.OrderDeliveryRequest;
import com.bubbleflow.dto.order.response.OrderResponse;
import com.bubbleflow.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface OrderService {
    PageResponse<OrderResponse> getAll(String search, String status, Pageable pageable);
    OrderResponse getById(Long id);
    OrderResponse create(OrderRequest request);
    OrderResponse updateStatus(Long id, String status);
    OrderResponse assignRack(Long id, Long rackId);
    OrderResponse deliverOrder(Long id, OrderDeliveryRequest request);
    List<OrderResponse> getSlaWarnings();
    void delete(Long id);
}

File: frontend/src/services/orderService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  PageResponse, 
  OrderResponse, 
  OrderRequest, 
  OrderDeliveryRequest,
  TableParams 
} from '@/types';

export const orderService = {
  getAll: (params?: TableParams & { status?: string }): Promise<ApiResponse<PageResponse<OrderResponse>>> => {
    return apiClient.get('/orders', { params });
  },

  getById: (id: number): Promise<ApiResponse<OrderResponse>> => {
    return apiClient.get(`/orders/${id}`);
  },

  create: (data: OrderRequest): Promise<ApiResponse<OrderResponse>> => {
    return apiClient.post('/orders', data);
  },

  updateStatus: (id: number, status: string): Promise<ApiResponse<OrderResponse>> => {
    return apiClient.put(`/orders/${id}/status`, null, { params: { status } });
  },

  assignRack: (id: number, rackId: number): Promise<ApiResponse<OrderResponse>> => {
    return apiClient.put(`/orders/${id}/assign-rack`, null, { params: { rackId } });
  },

  deliver: (id: number, data: OrderDeliveryRequest): Promise<ApiResponse<OrderResponse>> => {
    return apiClient.put(`/orders/${id}/delivery`, data);
  },

  getSlaWarnings: (): Promise<ApiResponse<OrderResponse[]>> => {
    return apiClient.get('/orders/sla-warnings');
  },

  notifyCustomer: (id: number): Promise<ApiResponse<OrderResponse>> => {
    return apiClient.put(`/orders/${id}/notify-customer`);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/orders/${id}`);
  },
};
